docs/cookbook.md

```diff
@@ -196,6 +196,71 @@ type::{
 }
 ```
 
+## Modeling SQL Decimals in Ion Schema
+
+Ion `decimal` and SQL `DECIMAL` have some fundamental differences that make it complex to compare values between the two types.
+
+For SQL `DECIMAL`, the precision of the number `12.34` is not determined by the actual number of digits, but (speaking informally) by the data type it is assigned. `12.34` can be assigned to `DECIMAL(38,2)` just as easily as `DECIMAL(4,2)`. That is to say, precision and scale are a property of the `DECIMAL` data _type_ rather than being inherent in the value.
+
+On the other hand, the Ion decimal data type has no particular precision or scale as it is an _arbitrary precision_ data type—the precision of an Ion decimal value is inherent in the value itself.
+
+As a result of these differences, when converting a value from a SQL `DECIMAL` to an Ion `decimal`, it is _possible_ to preserve the scale of the `DECIMAL` (as it becomes the exponent of the Ion decimal, multiplied by -1), but the precision is lost. When converting from an Ion `decimal` to a `DECIMAL(p,s)`, the precision and scale of the Ion value are always lost because the value is coerced into having the precision and scale of `DECIMAL(p,s)`.
+
+That being said, here are some questions that one might be trying to answer by modeling SQL `DECIMAL` and how to approach modeling them in Ion Schema.
+
+**Does `x` have the exact precision and the exact scale of `DECIMAL(p,s)`?**
+
+This is probably the least useful way to model a SQL `DECIMAL`, and is included here only to help explain why exact precision is not useful when modeling `DECIMAL` in Ion Schema.
+
+Using `DECIMAL(5,2)` as our example, we can model an _exact_ precision and scale like this:
+```
+type::{
+  precision: 5,
+  scale: 2,
+}
+```
+This will only allow `decimal`s with 5 digits, and two after the decimal point. However, this will not accept valid `DECIMAL(5,2)` values such as `1.00`.
+
+**Does `x` have a compatible precision and the exact scale of `DECIMAL(p,s)`?**
+
+This way of modeling a SQL `DECIMAL` is useful for scenarios where the value was a SQL `DECIMAL(p,s)` before being converted to Ion.
+
+This can be separated into two parts:
+1. Does `x` have exactly `s` digits after the decimal point?
+2. Does `x` have less than or equal to `p` digits? (Or is `-10^(p-s) < x < 10^(p-s)`?)
+
+
+Using `DECIMAL(5,2)` as our example, we can model a _compatible_ precision and _exact_ scale like this:
+```
+type::{
+  precision: range::[min, 5],
```

Comment:
According to[ Ion Schema Specification](https://amzn.github.io/ion-schema/docs/spec.html#precision), the `min` values of `precision` and `scale` are different. Should we add the clarification of the `min` value in this session?  For example: The minimum precision must be greater than or equal to 1. The minimum scale must be greater than or equal to 0.

Reply:
I think that is orthogonal to this change, and I don't think there is any confusion about the `min` value for `precision`.

The `min` value for scale has some other problems (mentioned in #27), and I think it would be best to address them separately; the (technically) incorrect implementation of `scale` in https://github.com/amzn/ion-schema-kotlin is actually necessary to be able to properly model SQL decimals.